examples/routed-echo/README.md

```diff
@@ -0,0 +1,49 @@
+# Routed Host: echo client/server
+
+This example is intended to follow up the basic host and echo examples by adding use of the ipfs distributed hash table to lookup peers.
+
+Functionally, this example works similarly to the echo example, however setup of the host includes wrapping it with a Kademila hash table, so it can find peers using only their ID. 
+
+## Build
+
+From `go-libp2p` base folder:
+
+```
+> make deps
+> go build ./examples/routed-echo
+```
+
+## Usage
+
+```
+> ./routed-echo -secio -l 10000
+2018/02/19 12:22:32 I can be reached at:
+2018/02/19 12:22:32 /ip4/127.0.0.1/tcp/10000/ipfs/QmfRY4vuKpU2tApACrbmYFn9xoeNzMQhLXg7nKnyvnzHeL
+2018/02/19 12:22:32 /ip4/192.168.1.203/tcp/10000/ipfs/QmfRY4vuKpU2tApACrbmYFn9xoeNzMQhLXg7nKnyvnzHeL
+2018/02/19 12:22:32 Now run "./routed-echo -l 10001 -d QmfRY4vuKpU2tApACrbmYFn9xoeNzMQhLXg7nKnyvnzHeL -secio" on a different terminal
+2018/02/19 12:22:32 listening for connections
+```
+
+The listener libp2p host will print its randomly generated Base58 encoded ID string, which combined with the ipfs DHT, can be used to reach the host, despite lacking other connection details.  You can also pass the `-global` flag to bootstrap using the same peers as ipfs.
```

Comment:
Maybe: "By default, this example will bootstrap off your local IPFS peer (assuming one is running). If you'd rather bootstrap off the same peers `go-ipfs` uses, pass the `-global` flag."

That is, something to indicate the default behaviour.

Reply:
note: I couldn't get it to work reliably if the first host was bootstrapped locally, and the second was bootstrapped globally.